daemon/containerd/leases.go

```diff
@@ -0,0 +1,51 @@
+package containerd
+
+import (
+	"context"
+	"time"
+
+	"github.com/containerd/containerd/leases"
+	"github.com/containerd/log"
+)
+
+const (
+	leaseExpireDuration = 8 * time.Hour
+	expireLabel         = "containerd.io/gc.expire" // Copied from containerd
+	pruneLeaseLabel     = "moby/prune.images"
+	pruneLeaseFilter    = `labels."moby/prune.images"`
+)
+
+func (i *ImageService) withLease(ctx context.Context, cancellable bool) (context.Context, func(), error) {
```

Comment:
Can you help me understand the reason one would set `cancellable` to true?

Reply:
Resumable pull